src/Compilers/CSharp/Test/Semantic/Semantics/StackAllocInitializerTests.cs

```diff
@@ -17,59 +17,86 @@ public class StackAllocInitializerTests : CompilingTestBase
         public void RestrictedTypesAllowedInStackalloc()
         {
             var comp = CreateCompilationWithMscorlibAndSpan(@"
-ref struct RefS { }
+public ref struct RefS { }
+public ref struct RefG<T> { public T field; }
 
 class C
 {
     unsafe void M()
     {
         var x1 = stackalloc RefS[10];
-        var x2 = stackalloc System.TypedReference[10];
-        var x3 = stackalloc System.ArgIterator[10];
-        var x4 = stackalloc System.RuntimeArgumentHandle[10];
+        var x2 = stackalloc RefG<string>[10];
+        var x3 = stackalloc RefG<int>[10];
+        var x4 = stackalloc System.TypedReference[10];
+        var x5 = stackalloc System.ArgIterator[10];
+        var x6 = stackalloc System.RuntimeArgumentHandle[10];
 
         var y1 = new RefS[10];
-        var y2 = new System.TypedReference[10];
-        var y3 = new System.ArgIterator[10];
-        var y4 = new System.RuntimeArgumentHandle[10];
+        var y2 = new RefG<string>[10];
+        var y3 = new RefG<int>[10];
+        var y4 = new System.TypedReference[10];
+        var y5 = new System.ArgIterator[10];
+        var y6 = new System.RuntimeArgumentHandle[10];
 
         RefS[] z1 = null;
-        System.TypedReference[] z2 = null;
-        System.ArgIterator[] z3 = null;
-        System.RuntimeArgumentHandle[] z4 = null;
+        RefG<string>[] z2 = null;
+        RefG<int>[] z3 = null;
+        System.TypedReference[] z4 = null;
+        System.ArgIterator[] z5 = null;
+        System.RuntimeArgumentHandle[] z6 = null;
         _ = z1;
         _ = z2;
         _ = z3;
         _ = z4;
+        _ = z5;
+        _ = z6;
     }
 }
 ", TestOptions.UnsafeReleaseDll);
 
             comp.VerifyDiagnostics(
-                // (13,22): error CS0611: Array elements cannot be of type 'RefS'
+                // (10,29): error CS0208: Cannot take the address of, get the size of, or declare a pointer to a managed type ('RefG<string>')
+                //         var x2 = stackalloc RefG<string>[10];
+                Diagnostic(ErrorCode.ERR_ManagedAddr, "RefG<string>").WithArguments("RefG<string>").WithLocation(10, 29),
+                // (11,29): error CS0208: Cannot take the address of, get the size of, or declare a pointer to a managed type ('RefG<int>')
+                //         var x3 = stackalloc RefG<int>[10];
```

Comment:
Heads up @JoeRobich, @agocke: this will cause a test failure when we merge to master as this will stop giving an error.

Reply:
Here's the commit to resolve that conflict in merge to `master`:
https://github.com/dotnet/roslyn/pull/34146/commits/7fa3083191ad25eff02d9a74cf35e58b2d41570d